skbio/sequence/_grammared_sequence.py

```diff
@@ -22,11 +23,71 @@
 from ._sequence import Sequence
 
 
-class IUPACSequence(with_metaclass(ABCMeta, Sequence)):
-    """Store biological sequence data conforming to the IUPAC character set.
+class GrammaredSequenceException(Exception):
```

Comment:
What about using `TypeError` for these exceptions? That'd match `ABCMeta` and `DisableSubclassingMeta` and I think makes sense (an error defining a new type). We've been preferring using stdlib error types over custom exceptions if there's one that fits.